### DISCUSSION TITLE
Adding additional pumps as switches

### DISCUSSION THREAD

**Discussion by richlscott** [jlvaillant/intellicenter#3] [Q&A]
First off,  thank you for making this great add-on.  It works great.

One question,  I have 3 total pumps.  1 for teh pool/spa, and two for waterfalls.  The waterfall pumps show up as binary sensors and not switches so I am unable to turn them on /off .   tried customizes the entities class to switch, but no luck.  Any ideas on how I can create the entires for these additional pumps ?

![image](https://user-images.githubusercontent.com/80927685/112574726-e331ec80-8dbc-11eb-8be5-6dafe6279df8.png)

**Comment by jlvaillant** [comment]
Thank you for the feedback :-) much appreciated...

Technically the panel does not treat the ‘pump objects’ as things you can control directly. For example, when the pump is shared between pool and spa, turning that pump on or off would be ambiguous.
Instead the panel publishes ‘circuits’ for such control. In your case I suspect there are 2 circuits representing the waterfalls (which themselves are controlling the pumps).
The integration only create Home Assistant switches for the ‘bodies’ (ie pool and spa), the lights and the circuits that are ‘featured’. 
Would you mind trying to go to the native IntelliCenter app (or website) and mark your waterfalls as ‘featured’ circuits and reload the integration?
I think that should solve your problem but if not I’ll be happy to keep digging into the issue.

**Comment by richlscott** [comment]
Boom,  that was it.

I had the two waterfalls put into a feature group which I controlled
together..  changing the individual pumps to feature made them available to
action as a switch.  Amazing.  I really appreciate your add-on.  thanks for
your contribution.

**Reply by jlvaillant** [reply]
Great! Glad it worked...

I must admit I did not think of that “use case”  of featured groups’ beside lights (not sure my pool has enough ‘gadgets’ for me to try that). I’ll look at the code and see if I find a way to deal with that as your ‘instinct’ to group the waterfalls in IntelliCenter makes sense to me. I know it’s trivial to group them in HomeAssistant but it’d be great for the integration to work ‘right of the box’...
